plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -402,31 +436,46 @@ fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
 						      error))
 		return FALSE;
 
-	/* prioritize DDC/CI commands in display controller */
-	if (!fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_UP, error))
+	/* get vid and pid from PCI bus */
+	udev_parent = fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), "pci");
+	if (udev_parent == NULL)
 		return FALSE;
 
-	/* mediatek display is connected */
-	if (!fu_mediatek_scaler_display_is_connected(device, error))
+	if (!fu_device_probe(FU_DEVICE(udev_parent), error))
 		return FALSE;
 
-	/* set hardware version */
-	hw_ver = fu_mediatek_scaler_device_get_hardware_version(device, error);
-	if (hw_ver == NULL)
+	vid_u16 = fu_udev_device_get_subsystem_vendor(udev_parent);
+	pid_u16 = fu_udev_device_get_subsystem_model(udev_parent);
+
+	/* supported host system only */
+	if (!(vid_u16 != 0x1028) && ((pid_u16 != 0x0c88) || (pid_u16 == 0x0c8a)))
 		return FALSE;
 
-	/* set vid and pid from PCI bus */
-	udev_parent = fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), "pci");
-	if (udev_parent == NULL)
+	/* supported controller type only */
+	if (!fu_mediatek_scaler_controller_in_display(device, error))
 		return FALSE;
 
-	if (!fu_device_probe(FU_DEVICE(udev_parent), error))
+	/* prioritize DDC/CI commands in display controller */
+	if (!fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_UP, error))
 		return FALSE;
 
-	vid = g_strdup_printf("%04X", fu_udev_device_get_subsystem_vendor(udev_parent));
-	pid = g_strdup_printf("%04X", fu_udev_device_get_subsystem_model(udev_parent));
+	/* mediatek display is connected */
+	if (!fu_mediatek_scaler_display_is_connected(device, error)) {
+		fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_NORMAL, error);
```

Comment:
in contrast to add this in every false returns, any better way to run a code when probe leaves?

Reply:
I think we want to clean it up something like what's in wip/hughsie/mediatek-weird -- you're relying on a lot of daemon auto-fixup at the moment, e.g. refcounting the opens of the i2c device for instance.

Reply:
Let me rebase this onto  wip/hughsie/mediatek-weird -- then I think it'll be a lot nicer.

Reply:
@CragW I've merged that functionality on top of wip/hughsie/mediatek-weird -- can you test that branch for me please.